avm/utilities/pipelines/sharedScripts/Get-BRMRepositoryName.ps1

```diff
@@ -21,6 +21,6 @@ function Get-BRMRepositoryName {
         [string] $TemplateFilePath
     )
 
-    $moduleIdentifier = (Split-Path $TemplateFilePath -Parent) -split '[\/|\\](avm[\/|\\])'
-    return ($moduleIdentifier[1, 2] -join '') -replace '\\', '/'
+    $moduleIdentifier = (Split-Path $TemplateFilePath -Parent) -split '[\/|\\]avm[\/|\\](res|ptn)[\/|\\]'
```

Comment:
As an additional measure, I'd suggest throwing an error if templateFilePath contains the separator avm(ptn|res) twice

Reply:
Thanks for the comment @eriqua. Do I understand this right that you'd want to catch cases like "/avm/res/*/res/"?  Is that a likely scenario?
And if you say it is, should I then add this test in all locations where we split? I guess testing would add a few lines each as I'd need to search for multiple regex tuples

Reply:
It's very unlikely but possible. If requires additional effort let's park it for now. Instead, we may want to think of getting the repo relative path aa a shared function, instead of replacing the same in multiple locations. Not urgent for now anyway.

Reply:
Hey @eriqua, honestly I don't think this path is in any way a realistic scenario and would hence very much like to park this thought as 'fixing it' would in turn also require us to fix it in 5 places (like you pointed out yourself). 😨

Reply:
Again, fair.